packages/plugin-autocapture-browser/src/hierarchy.ts

```diff
@@ -0,0 +1,135 @@
+import { isNonSensitiveElement } from './helpers';
+import { Hierarchy, HierarchyNode } from './typings/autocapture';
+
+const BLOCKED_ATTRIBUTES = [
+  // Already captured elsewhere in the hierarchy object
+  'id',
+  'class',
+
+  // non-useful and potentially large attribute
+  'style',
+
+  // sensitive as prefilled form data may populate this attribute
+  'value',
+
+  // DOM events
+  'onclick',
+  'onchange',
+  'oninput',
+  'onblur',
+  'onsubmit',
+  'onfocus',
+  'onkeydown',
+  'onkeyup',
+  'onkeypress',
+
+  // React specific
+  'data-reactid',
+  'data-react-checksum',
+  'data-reactroot',
+];
+const SENSITIVE_ELEMENT_ATTRIBUTE_ALLOWLIST = ['type'];
+
+const SVG_TAGS = ['svg', 'path', 'g'];
+const HIGHLY_SENSITIVE_INPUT_TYPES = ['password', 'hidden'];
+const MAX_ATTRIBUTE_LENGTH = 128;
+const MAX_HIERARCHY_LENGTH = 1024;
+
+export function getElementProperties(element: Element | null): HierarchyNode | null {
+  if (element === null) {
+    return null;
+  }
+
+  const tagName = element.tagName.toLowerCase();
+  const properties: HierarchyNode = {
+    tag: tagName,
+  };
+
+  const siblings = Array.from(element.parentElement?.children ?? []);
+  if (siblings.length) {
+    properties.index = siblings.indexOf(element);
+    properties.indexOfType = siblings.filter((el) => el.tagName === element.tagName).indexOf(element);
```

Comment:
Curious what are these 2 properties used for? Is it for the query engine to match back the node?

Reply:
The information from these are used to enable nth-child and nth-of-type queries